src/autoloads/characters.gd

```diff
@@ -155,34 +158,92 @@ func _process(delta: float) -> void:
 
 # puppet keyword means that when this function is used in an rpc call
 # 	it will only be run on client
-puppet func _updateAllCharacterPositions(positions: Dictionary) -> void:
+puppet func _updateAllCharacterPositions(positions: Dictionary, characterData: Array) -> void:
+	var myId: int = get_tree().get_network_unique_id()
 	## Loop through all characters
 	for characterId in positions:
 		# if this position is for this client's character
-		if characterId == get_tree().get_network_unique_id():
+		if characterId == myId:
 			# don't update its position
 			continue
 		## Set the position for the character
 		getCharacterResource(characterId).setPosition(positions[characterId])
-
-# --Server Functions--
+	## Decompose character data
+	if len(characterData) > 0:
+		print_debug(characterData)
+	for data in characterData:
+		if data["to"] == myId or data["to"] == -1:
+			receiveCharacterDataClient(data["id"], data["data"])
+
+func sendCharacterData() -> void:
+	var id: int = get_tree().get_network_unique_id()
+	var characterRes: CharacterResource
+	characterRes = Characters.getCharacterResource(id)
+	var characterData: Dictionary = {}
+	characterData["outfit"] = characterRes.getOutfit()
+	characterData["colors"] = characterRes.getColors()
+	broadcastDataQueue.append(characterData)
+
+puppet func receiveCharacterDataClient(id: int, characterData: Dictionary) -> void:
+	var gameScene: Node = TransitionHandler.gameScene
+	gameScene.setCharacterData(id, characterData)
+
+## --Server Functions--
+
+master func sendAllCharacterData() -> void:
+	var characterRes: Dictionary = {}
+	characterRes = getCharacterResources()
+	for player in characterRes:
+		var characterData: Dictionary = {}
+		characterData["outfit"] = characterRes[player].getOutfit()
+		characterData["colors"] = characterRes[player].getColors()
+		var dataSend: Dictionary = {"to": -1, "id": player, "data": characterData}
+		broadcastDataQueue.append(dataSend)
+	print_debug(characterRes)
+	print_debug(broadcastDataQueue)
+
+master func receiveCharacterDataServer(characterData: Array) -> void:
+	var senderId: int = get_tree().get_rpc_sender_id()
+	var gameScene: Node2D = TransitionHandler.gameScene
+	## Decompose and compile received data
+	if len(characterData) == 0:
+		return
+	var compiledData: Dictionary = {}
+	for element in characterData:
+		for key in element:
+			compiledData[key] = element[key]
+	# Here the server could check and modify the data if necessary
+	## Sets character data for the character requested
+	gameScene.setCharacterData(senderId, compiledData)
+	## Broadcast new data
+	var dataSend: Dictionary = {"to": -1, "id": senderId, "data": compiledData}
+	broadcastDataQueue.append(dataSend)
 
 # receive a client's position
 # master keyword means that this function will only be run on the server when RPCed
-master func _receiveCharacterPosFromClient(newPos: Vector2) -> void:
+master func _receiveCharacterPosFromClient(newPos: Vector2, characterData: Array) -> void:
```

Comment:
Same thing, this should also be renamed to show that it is also sending character data

Reply:
Good call.